pkg/netpol/eval/resources.go

```diff
@@ -157,30 +163,35 @@ func (pe *PolicyEngine) addObjectsByKind(objects []parser.K8sObject) error {
 func (pe *PolicyEngine) resolveMissingNamespaces() error {
 	for _, pod := range pe.podsMap {
 		ns := pod.Namespace
-		if _, ok := pe.namspacesMap[ns]; !ok {
-			if err := pe.resolveSingleMissingNamespace(ns, nil); err != nil {
-				return err
-			}
+		if err := pe.addIfMissingNamespace(ns); err != nil {
+			return err
 		}
 	}
 	return nil
 }
 
-// resolveSingleMissingNamespace create a ns object and upsert to PolicyEngine
-func (pe *PolicyEngine) resolveSingleMissingNamespace(ns string, nsLabels map[string]string) error {
-	nLabels := nsLabels
-	if len(nLabels) == 0 {
-		nLabels = map[string]string{
-			common.K8sNsNameLabelKey: ns,
-		}
+// addMissingNamespace - helping func (to avoid duplicates); checks if the ns is not in the namespacesMap, resolves it
+func (pe *PolicyEngine) addIfMissingNamespace(ns string) (err error) {
```

Comment:
this func can be part of the `resolveSingleMissingNamespace` func?

Reply:
yes, done